x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java

```diff
@@ -79,34 +81,83 @@ protected void masterOperation(
             EnterpriseSearchFeatureSetUsage usage = new EnterpriseSearchFeatureSetUsage(
                 LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
                 enabled,
+                Collections.emptyMap(),
                 Collections.emptyMap()
             );
             listener.onResponse(new XPackUsageFeatureResponse(usage));
             return;
         }
 
-        try {
-            ListSearchApplicationAction.Response resp = clientWithOrigin.execute(
-                ListSearchApplicationAction.INSTANCE,
-                new ListSearchApplicationAction.Request(null, new PageParams(0, 0))
-            ).get();
+        Map<String, Object> searchApplicationsUsage = new HashMap<>();
+        Map<String, Object> analyticsCollectionsUsage = new HashMap<>();
+
+        // Step 2: Fetch search applications count and return usage
```

Comment:
nitpick -should this be step 1? and should we also change the other comment `// Step 1: Fetch analytics collections count`.

Reply:
these are defined in reverse order so they can be chained. this is why i've added the comments so the actual execution order is clearer.

would you like me to change the order or execution?

Reply:
makes sense - the diff confused me a bit, found it easier to look at the entire file